homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+import godice
+
+from homeassistant import const
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DiceConfigEntry
+from .const import DICE_MAX_NUMBER, DICE_MIN_NUMBER, DOMAIN, MANUFACTURER, MODEL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number",
+    device_class=SensorDeviceClass.ENUM,
+    options=[str(num) for num in range(DICE_MIN_NUMBER, DICE_MAX_NUMBER + 1)],
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name.lower() for color in godice.Color],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
```

Comment:
Okay so the main design idea behind entity descriptions is that you have 1 generic entity, where you change the data it exposes and how that data is exposed depending on the description that is injected. Currently you have 3 separate classes with different behaviours so I would say that we are not using the entity descriptions optimally. And looking at the 3 different classes, they differ quite a lot, so maybe entity descriptions just isn't a good fit and you can set everything as a shorthand attribute `_attr_device_class = ...` in the class it should be.